xarray/core/indexing.py

```diff
@@ -616,12 +635,18 @@ def __getitem__(self, indexer):
         self._check_and_raise_if_non_basic_indexer(indexer)
         return type(self)(self.array, self._updated_key(indexer))
 
+    def _vindex_set(self, key, value):
+        raise NotImplementedError(
+            "Lazy item assignment with the vectorized indexer is not yet "
+            "implemented. Load your data first by .load() or compute()."
+        )
+
+    def _oindex_set(self, key, value):
+        full_key = self._updated_key(key)
+        self.array[full_key] = value
```

Comment:
```suggestion
        self.array.oindex[full_key] = value
```

This should be failing now? Let's add a test for this path if we don't do it already?